api/server/router/network/network_routes.go

```diff
@@ -79,8 +80,16 @@ func (n *networkRouter) postNetworkCreate(ctx context.Context, w http.ResponseWr
 		return err
 	}
 
-	if _, err := n.clusterProvider.GetNetwork(create.Name); err == nil {
-		return libnetwork.NetworkNameError(create.Name)
+	if nw, err := n.clusterProvider.GetNetwork(create.Name); err == nil {
```

Comment:
> isn't it the same as partial ID matches for containers; ID > full name match > partial ID match

I am not sure. Typically, there should be a resolution component and the getter should _only_ take ID. In this case, I think we hit the docker API, so ambiguity of id/name may be a problem.